File: observer/php/demo1/Subject.php
<?php
namespace DesignPattern\ObServer;

class Subject
{
    private $_observer = [];

    public function attach(ObServer $observer)
    {
        array_push($this->_observer, $observer);
    }

    public function dettach(ObServer $observer)
    {
        if ($index = array_search($observer, $this->_observer)) {
            unset($this->_observer[$index]);
        }
    }

    public function notify($content)
    {
        foreach ($this->_observer as $observer) {
            $observer->update($content);
        }
    }
}

File: observer/java/demo1/Subject.java
import java.util.ArrayList;
import java.util.List;

/**
 * 目标对象
 */
public class Subject
{
    /**
     * 保存注册的观察者对象
     */
    private List<Observer> readers = new ArrayList<Observer>();

    /**
     * 注册观察者
     */
    public void attach(Observer reader){
        readers.add(reader);
    }

    /**
     * 取消观察者
     */
    public void detach(Observer reader){
        readers.remove(reader);
    }

    /**
     * 通知观察者
     */
    protected void notifyObservers() {
        for (Observer reader : readers ) {
            reader.update(this);
        }
    }
}